app/policies/wiki_policy.rb

```diff
@@ -1,4 +1,22 @@
 class WikiPolicy < ApplicationPolicy
-    def index?
-    end
+    # class Scope
+    #     attr_reader :user, :scope
+        
+    #     def initialize(user, scope)
+    #         @user  = user
+    #         @scope = scope
+    #     end
+        
+    #     def resolve
+    #         if user.admin? || user.premium?
+    #             scope.all
+    #         else
+    #             scope.where(private: false)
+    #         end
+    #     end
+    # end
+    
+    # def index?
+    #     user.admin? || user.premium? or not wiki.private?
```

Comment:
Don't mix and match your or styles here. Generally, we don't write a policy for index? -  we just use the resolve method, which gets called when you call policy_scope Wiki. 

Also - I'll need to see your ApplicationPolicy. Since you commented everything out of this file, that's what's driving it.